src/esmockIsLoader.js

```diff
@@ -1,5 +1,2 @@
-const isloaderRe = /--(experimental-)?loader[=\s,"']*esmock/
-
-export default (pr = process) =>
-  isloaderRe.test(pr.execArgv) ||
-  isloaderRe.test(pr.env.NODE_OPTIONS)
+export default (cache => async () => typeof cache === 'boolean' ? cache
+  : cache = ((await import(`${import.meta.url}?test`)).default === true))()
```

Comment:
good observation... I want to avoid any errors that may occur when multiple loaders are used and moduleId does not exist in file system. What do you think of something like this instead?
```javascript
await import(`${import.meta.url}?esmock-loader-test`)
```
then inside the loader, check the query string only,
```javascript
if (url.endsWith('?esmock-loader-test')) { ... }
```

Reply:
In an ideal world (and by specs) loaders mustn't touch anything that they do not know. But yes, somebody can write very useful but greedy loader which fails here.

```ts
import { verySpecialQueryString } from './esmockLoader.js'
...
await import(`${import.meta.url}?${verySpecialQueryString}`)
```

Reply:
this area is changed and improved (imo)
esmockLoader.js
```javascript
const loaderVerificationQuery = 'esmock-loader=true'
const loaderVerificationURLCreate = url => `${url}?${loaderVerificationQuery}`
const loaderIsVerified = async url =>
  (await import(loaderVerificationURLCreate(url))).default === true
const load = async (url, context, nextLoad) => {
  if (url.endsWith(`esmockIsLoader.js?${loaderVerificationQuery}`)) {
    ...
```
esmockIsLoader.js
```javascript
import { loaderIsVerified } from './esmockLoader.js'

export default (cache => async () => typeof cache === 'boolean' ? cache
  : cache = await loaderIsVerified(import.meta.url))()
```
what do you think?

Reply:
it's better - less points to guess why

Reply:
one last change...
```diff
-  if (url.endsWith(`esmockIsLoader.js?${loaderVerificationQuery}`)) {
+  if (url.endsWith(loaderVerificationQuery)) {
```